plugin/code_actions.py

```diff
@@ -12,37 +12,64 @@
 from .core.diagnostics import get_point_diagnostics
 from .core.url import filename_to_uri
 from .core.views import region_to_range
-from .core.helpers import debounce
 from .core.registry import session_for_view
 from .core.settings import settings
 
 
-class CodeAction:
-    def __init__(self, view: 'sublime.View') -> None:
-        self.view = view
+def send_code_action_request(view, on_response_recieved: 'Callable'):
+    session = session_for_view(view)
+    if not session or not session.has_capability('codeActionProvider'):
+        # the server doesn't support code actions, just return
+        return
+
+    region = view.sel()[0]
+    pos = region.begin()
+    point_diagnostics = get_point_diagnostics(view, pos)
+    params = {
+        "textDocument": {
+            "uri": filename_to_uri(view.file_name())
+        },
+        "range": region_to_range(view, region).to_lsp(),
+        "context": {
+            "diagnostics": list(diagnostic.to_lsp() for diagnostic in point_diagnostics)
+        }
+    }
+    session.client.send_request(
+        Request.codeAction(params),
+        lambda response: on_response_recieved(response))
```

Comment:
This turned out to be a really nice cleanup!

Reply:
Thanks :D